lib/gitsh/tab_completion/dsl/option_transition_factory.rb

```diff
@@ -0,0 +1,60 @@
+require 'gitsh/tab_completion/automaton'
+require 'gitsh/tab_completion/matchers/unknown_option_matcher'
+
+module Gitsh
+  module TabCompletion
+    module DSL
+      class OptionTransitionFactory
+        def build(start_state, options = {})
+          Invocation.new(start_state, options).call
```

Comment:
Perhaps not. I'm not a big fan of setting ivars outside of `#initialize` (unless they're memoising a lazily calculated value), but the inner class makes this harder to read.